tests/unit/regression/lars_unittest.cc

```diff
@@ -91,6 +91,30 @@ void generate_data_n_less_d(SGMatrix<float64_t> &data, SGVector<float64_t> &lab)
 	lab[2]=-0.553422897279663;
 }
 
+//Generate the Data that N is less than D
+void generate_data_n_less_d(SGMatrix<float32_t> &data, SGVector<float64_t> &lab)
+{
```

Comment:
Can't you re-use the existing code in the sense that you generate the float64 data, and then manually copy it elementwise to float32 structures?